addons/sms/wizard/sms_compose_message.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError
+from odoo.addons.iap.models import iap
+from odoo.addons.mail.wizard.mail_compose_message import _reopen
+
+_logger = logging.getLogger(__name__)
+
+class SMSRecipient(models.TransientModel):
+    _name = 'sms.recipient'
+    _description = 'SMS Recipient'
+
+    sms_composer_id = fields.Many2one('sms.compose.message')
+    partner_id = fields.Many2one('res.partner')
+    number = fields.Char(required=True)
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        for record in self:
+            record.number = record.partner_id.mobile if record.partner_id.mobile else record.partner_id.phone
+
+class SendSMS(models.TransientModel):
+    _name = 'sms.compose.message'
+    _description = 'Send SMS Wizard'
+
+    composition_mode = fields.Selection([('comment', 'Post on a document'), ('mass_sms', 'SMS Mass Mailing')], default='comment')
+    use_active_domain = fields.Boolean()
+
+    recipient_ids = fields.One2many('sms.recipient', 'sms_composer_id', string='Recipients')
+    model = fields.Char()
+    template_id = fields.Many2one('sms.template', string='Use Template', domain="[('model', '=', model)]")
+    content = fields.Text(required=True)
+
+    @api.onchange('template_id')
+    def _onchange_template_id(self):
+        if self.template_id:
+            self.content = self.template_id.body
+
+    def _get_records(self, model):
+        if self.use_active_domain or self.env.context.get('active_domain'):
+            records = model.search(self.env.context.get('active_domain'))
+        elif self.env.context.get('active_ids'):
+            records = model.browse(self.env.context.get('active_ids', []))
+        else:
+            records = model.browse(self.env.context.get('active_id', []))
+        return records
+
+    @api.model
+    def default_get(self, fields):
+        result = super(SendSMS, self).default_get(fields)
+        active_model = self.env.context.get('active_model')
+        result['model'] = active_model
+        model = self.env[active_model]
+        records = self._get_records(model)
+        if self.env.context.get('default_composition_mode') != 'mass_sms' \
+            and not self.env.context.get('default_recipient_ids'):
+            recipients = self.env['sms.sms']._get_sms_recipients(active_model, records.id)
+            missing_numbers = []
```

Comment:
missing_numbers = [recipient['partner_id'].display_name for recipient in recipients if not recipient['number']]